File: py-opsml/python/opsml/genai/google.py
# type: ignore
# pylint: disable=no-member
# opsml/genai/google.py
from . import google as _google_impl

Modality = _google_impl.Modality
ThinkingConfig = _google_impl.ThinkingConfig
MediaResolution = _google_impl.MediaResolution
SpeechConfig = _google_impl.SpeechConfig
PrebuiltVoiceConfig = _google_impl.PrebuiltVoiceConfig
VoiceConfigMode = _google_impl.VoiceConfigMode
VoiceConfig = _google_impl.VoiceConfig
GenerationConfig = _google_impl.GenerationConfig
HarmCategory = _google_impl.HarmCategory
HarmBlockThreshold = _google_impl.HarmBlockThreshold
HarmBlockMethod = _google_impl.HarmBlockMethod
SafetySetting = _google_impl.SafetySetting
ToolConfig = _google_impl.ToolConfig
FunctionCallingConfig = _google_impl.FunctionCallingConfig
RetrievalConfig = _google_impl.RetrievalConfig
LatLng = _google_impl.LatLng
ModelArmorConfig = _google_impl.ModelArmorConfig
Mode = _google_impl.Mode
GeminiSettings = _google_impl.GeminiSettings
GeminiEmbeddingConfig = _google_impl.GeminiEmbeddingConfig
GeminiEmbeddingResponse = _google_impl.GeminiEmbeddingResponse
PredictRequest = _google_impl.PredictRequest
PredictResponse = _google_impl.PredictResponse
EmbeddingTaskType = _google_impl.EmbeddingTaskType

__all__ = [
    "Modality",
    "ThinkingConfig",
    "MediaResolution",
    "SpeechConfig",
    "PrebuiltVoiceConfig",
    "VoiceConfigMode",
    "VoiceConfig",
    "GenerationConfig",
    "ToolConfig",
    "FunctionCallingConfig",
    "RetrievalConfig",
    "LatLng",
    "ModelArmorConfig",
    "Mode",
    "GeminiSettings",
    "HarmCategory",
    "HarmBlockThreshold",
    "HarmBlockMethod",
    "SafetySetting",
    "GeminiEmbeddingConfig",
    "GeminiEmbeddingResponse",
    "PredictRequest",
    "PredictResponse",
    "EmbeddingTaskType",
]

File: py-opsml/src/genai/google.rs
use potato_head::google_types::{
    chat::{
        FunctionCallingConfig, GeminiSettings, GenerationConfig, HarmBlockMethod,
        HarmBlockThreshold, HarmCategory, LatLng, MediaResolution, Modality, Mode,
        ModelArmorConfig, PrebuiltVoiceConfig, RetrievalConfig, SafetySetting, SpeechConfig,
        ThinkingConfig, ToolConfig, VoiceConfig, VoiceConfigMode,
    },
    predict::{PredictRequest, PredictResponse},
    EmbeddingTaskType, GeminiEmbeddingConfig, GeminiEmbeddingResponse,
};
use pyo3::prelude::*;

pub fn add_google_module(m: &Bound<'_, PyModule>) -> PyResult<()> {
    m.add_class::<GenerationConfig>()?;
    m.add_class::<MediaResolution>()?;
    m.add_class::<Modality>()?;
    m.add_class::<SpeechConfig>()?;
    m.add_class::<ThinkingConfig>()?;
    m.add_class::<VoiceConfig>()?;
    m.add_class::<VoiceConfigMode>()?;
    m.add_class::<PrebuiltVoiceConfig>()?;
    m.add_class::<HarmCategory>()?;
    m.add_class::<HarmBlockMethod>()?;
    m.add_class::<HarmBlockThreshold>()?;
    m.add_class::<SafetySetting>()?;
    m.add_class::<ToolConfig>()?;
    m.add_class::<FunctionCallingConfig>()?;
    m.add_class::<RetrievalConfig>()?;
    m.add_class::<LatLng>()?;
    m.add_class::<ModelArmorConfig>()?;
    m.add_class::<Mode>()?;
    m.add_class::<GeminiSettings>()?;
    m.add_class::<GeminiEmbeddingConfig>()?;
    m.add_class::<GeminiEmbeddingResponse>()?;
    m.add_class::<PredictRequest>()?;
    m.add_class::<PredictResponse>()?;
    m.add_class::<EmbeddingTaskType>()?;
    Ok(())
}
